src/blocks/ui/block.js

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import Cards from './components/cards';
+import DateRange from './components/date-range';
+import Variants from './components/variants';
+
+const { Icon } = wp.components;
+const { __ } = wp.i18n;
+const { useSelect } = wp.data;
+
+const BlockWrapper = styled.div`
+	padding: 40px 60px;
+	margin-left: -20px;
+
+	& h1,
+	& h2 {
+		font-size: 24px;
+		margin: 10px 0;
+		line-height: 1;
+	}
+	& h1 + h2 {
+		color: #4767df;
+	}
+
+	& h2 a {
+		text-decoration: none;
+		border-radius: 3px;
+		margin-left: 4px;
+		margin-top: -4px;
+		border: 1px solid rgba(67, 108, 255, .4);
+		width: 20px;
+		height: 20px;
+		display: inline-block;
+		vertical-align: middle;
+		padding: 0;
+	}
+
+	.altis-analytics-block-metrics {
+		margin: 40px -60px;
+		background-color: rgba(67, 108, 255, .05);
+		padding: 20px 60px;
+	}
+
+	.altis-analytics-date-range {
+		margin: 0 0 20px;
+	}
+`;
+
+/**
+ * Experience Block Analytics component.
+ *
+ * @param {object} props The component props.
+ * @param {string} props.clientId The block client ID.
+ * @param {boolean} props.showPosts Whether to show posts this XB is shown on.
+ * @returns {React.ReactNode} The block view component.
+ */
+const Block = ( {
+	clientId,
+} ) => {
+	const [ days, setDays ] = useState( 7 );
+	const block = useSelect( select => {
+		return select( 'analytics/xbs' ).getPost( clientId );
+	}, [ clientId ] );
+	const analytics = useSelect( select => {
+		return select( 'analytics/xbs' ).getViews( clientId, { days } );
+	}, [ clientId, days ] );
+	const lift = useSelect( select => {
+		const current = select( 'analytics/xbs' ).getViews( clientId, { days: 7 } );
+		const previous = select( 'analytics/xbs' ).getViews( clientId, {
+			days: 7,
+			offset: 7,
+		} );
+		return {
+			current,
+			previous,
+		};
+	}, [ clientId ] );
+
+	// Ensure we have a block ID data.
+	if ( ! clientId ) {
+		return (
+			<div className="message error">
+				<p>{ __( 'Experience Block not found.' ) }</p>
+			</div>
+		);
+	}
+	if ( block && block.error ) {
+		return (
+			<div className="message error">
+				<p>{ block.error.message }</p>
+			</div>
+		);
+	}
+
+	// Get percentage of personalised block views.
+	let personalisedCoverage = null;
+	if ( analytics && analytics.audiences.length > 0 ) {
+		const fallback = analytics.audiences.find( audience => audience.id === 0 );
+		personalisedCoverage = 100 - ( ( fallback.unique.views / analytics.unique.views ) * 100 );
+	}
+
+	return (
+		<BlockWrapper className="altis-analytics-block">
+			<h1>{ __( 'Experience Insights', 'altis-analytics' ) }</h1>
+			<h2>
+				{ ( block && block.title.rendered ) || __( 'Loading…', 'altis-analytics' ) }
+				{ block && block.parent && (
```

Comment:
This is returning `0` for me, so my title looks like:
```html
<h2>
    XB Analytics Demo
    0
</h2>
```

![Screen Shot 2021-03-15 at 12 12 24 PM](https://user-images.githubusercontent.com/991511/111200809-bd6c4280-8587-11eb-871c-3cd3a15c21e9.png)

We should fail gracefully so a failed conditional doesn't output in the markup.

Reply:
Oh derp, should be an expression that returns a bool when using that pattern
```suggestion
				{ block && block.parent > 0 && (
```